src_code/src/controllers/services/Authentication/SignUp.js

```diff
@@ -0,0 +1,163 @@
+            LAST_NAME: lastName.trim(),
+            STUDENT_ID: req.studentId,
+            EMAIL_ADDRESS: emailAddress.trim(),
+            ACCOUNT_PASSWORD: encryptedPassword,
+        }
+
+        /** Inserting 'insertData' into the table */
+        await db("user_info").insert(insertData);
+
+        /** Return 'Sign up successfully' message after insert to table successfully */
+        return responseBuilder.BuildResponse(res, 200, {
+            message: "Sign up successfully."
+        });
+    } catch(error){ 
+        /** If an unexpected server error occurs while signing up after passing validation, return log and error. */
+        console.log("ERROR: An error occurred while signing up: ", error);
+        return responseBuilder.ServerError(res, "Sorry, there was an error while signing up.");
+    }
+}
+
+/**
+ * Handle sign up validation (TODO: Have not verify if email address is faculty or admin yet).
+ * @param {object} res - the response object for the HTTP request.
+ * @param {object} req - the body of the request containing sign up details.
+ * @returns {object} res - if failed validation, send back to client bad request response, otherwise null.
+ */
+async function SignUpValidation(res, req) {
+    try {
+        /** Declare a email pattern ends with @spu.edu (regex) */
+        const emailPattern = /.+@spu\.edu$/;
+
+        /** Declare a studentID pattern */
+        const studentIDPattern = /^900\d{6}$/;
+
+
+        /** Destructure variables from the request body */
+        const {userRole, firstName, lastName, studentId, emailAddress, accountPassword } = req;
+
+        /** We check for userRole and studentId even though they are optional because we expect to have them in request body */
+        if(!userRole || !firstName || !lastName || !emailAddress || !accountPassword) {
+            return responseBuilder.MissingContent(res);
+        }
+
+        if(userRole.trim() !== "Student" && userRole.trim() !== "Faculty" && userRole.trim() !== "Admin") {
+            return responseBuilder.BadRequest(res, "Invalid user role.")
+        }
+
+        if(firstName.length > 25) {
+            return responseBuilder.BadRequest(res, "First name is longer than 25 characters.");
+        }
+
+        if(lastName.length > 25) {
+            return responseBuilder.BadRequest(res, "Last name is longer than 25 characters.");
+        }
+
+        if(req.studentId && req.studentId.length > 0){
+            /** Verifying if the student id is valid */
+            if(req.studentId.length !== 9 || !studentIDPattern.test(req.studentId)) {
+                return responseBuilder.BadRequest(res, "Invalid student ID.");
+            }
+
+            /** Retrieve user with the sign up student id to check for the uniqueness */
+            const isUserWithRequestStudentId = await dbHelper.GetUserInfoByStudentId(db, studentId.trim());
+            
+            if(isUserWithRequestStudentId && typeof isUserWithRequestStudentId === "string") {
```

Comment:
What does this kind of conditional mean again?

Reply:
done add comments! I forgot this line